### PR TITLE
Add upload request to TesterApiHttpClient

### PR DESCRIPTION
**Pull request by lfkellogg**
This will be used to upload screenshots. An upload request is just a POST to a different host with some extra headers.

### CODE REVIEW

File: firebase-appdistribution/src/main/java/com/google/firebase/appdistribution/impl/TesterApiHttpClient.java [hunk 1]
```diff
@@ -48,6 +52,11 @@ class TesterApiHttpClient {
   private static final String X_ANDROID_CERT_HEADER_KEY = "X-Android-Cert";
   // Format of "X-Client-Version": "{ClientId}/{ClientVersion}"
   private static final String X_CLIENT_VERSION_HEADER_KEY = "X-Client-Version";
+  private static final String X_GOOG_UPLOAD_PROTOCOL_HEADER = "X-Goog-Upload-Protocol";
+  private static final String X_GOOG_UPLOAD_PROTOCOL_RAW = "raw";
+  private static final String X_GOOG_UPLOAD_FILE_NAME_HEADER = "X-Goog-Upload-File-Name";
+  private static final String X_GOOG_UPLOAD_FILE_NAME = "screenshot.png";
+  private static final String UTF_8 = "UTF-8";
```
Comment (kaibolay) [inline firebase-appdistribution/src/main/java/com/google/firebase/appdistribution/impl/TesterApiHttpClient.java:59]: why not use `StandardCharsets.UTF_8`?

File: firebase-appdistribution/src/main/java/com/google/firebase/appdistribution/impl/TesterApiHttpClient.java [hunk 2]
```diff
@@ -35,7 +38,8 @@
 /** Client that makes FIS-authenticated GET and POST requests to the App Distribution Tester API. */
 class TesterApiHttpClient {
 
-  @VisibleForTesting static final String APP_TESTERS_HOST = "firebaseapptesters.googleapis.com";
+  private static final String APP_TESTERS_HOST = "firebaseapptesters.googleapis.com";
+  private static final String APP_TESTERS_UPLOAD_HOST = "firebaseapptesters.clients6.google.com";
```
[the remark this answers is not in the document]
Comment (rachaprince) [reply firebase-appdistribution/src/main/java/com/google/firebase/appdistribution/impl/TesterApiHttpClient.java:42]: +1

[the remark this answers is not in the document]
Comment (lfkellogg) [reply firebase-appdistribution/src/main/java/com/google/firebase/appdistribution/impl/TesterApiHttpClient.java]: For the life of me I can't figure out why I thought we needed a different host for this...

File: firebase-appdistribution/src/main/java/com/google/firebase/appdistribution/impl/TesterApiHttpClient.java (hunk 1 above)
Comment (lfkellogg) [reply firebase-appdistribution/src/main/java/com/google/firebase/appdistribution/impl/TesterApiHttpClient.java:59]: It's not available unfortunately: `Field requires API level 19 (current min is 16): java.nio.charset.StandardCharsets#UTF_8`. I added a comment.

Apparently I can use it in a test though.

**Review by kaibolay — APPROVED** [review]

**Pull request merged**